### PR TITLE
Update error catching with the correct exceptions

### PR DESCRIPTION
**Pull request by betrevisan**
The changes proposed in this pull request update the error catching with the appropriate exceptions instead of trying to catch all exceptions.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 1]
```diff
@@ -37,9 +37,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
 
-            String where = (String) userInfoEntity.getProperty("where");
-            // If there is nothing in the where filed, set it to contact.html by default.
-            if (where == null) {
+            String where = "/contact.html";
+            try {
+                where = (String) userInfoEntity.getProperty("where");
+            } catch (NullPointerException e) {
+                // If there is nothing in the where filed, set it to contact.html by default.
                 where = "/contact.html";
```
Comment (demlian) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:45]: this a Java thing? is the variable scope reduced to the try/catch block?

you're initializing a "where" variable in line 40 then redefining it in line 45

Comment (betrevisan) [reply portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:45]: Yes, if I declared the variable inside try I would only be able to access it inside the try. So I could access it later in the code, I declared the variable before the try-catch and then just updated its values inside that.

Comment (demlian) [reply portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:45]: give this a try @ https://www.tutorialspoint.com/compile_java_online.php

```
public class HelloWorld{

     public static void main(String []args){
        System.out.println("Hello World");
            String where = "/contact.html";
            try {
                System.out.println(where);
                throw new NullPointerException();
            } catch (NullPointerException e) {
                 System.out.println(where);
            }
     }
}
```

Comment (betrevisan) [reply portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:45]: I addressed this inside #57

**Pull request merged**